### ISSUE TITLE
token-cli: Unable to use the token cli to transfer tokens that have the transfer-fee and transfer-hook extensions. 

### ISSUE THREAD

**Issue by tonton-sol** [solana-labs/solana-program-library#7059]
I am trying to test using the transfer-fee extension in conjunction with the transfer-hook extension. The transfer-hook is the provided example program given in this repository. The transfer hook extra account metas account is initialized but empty. I am running this command:

```bash
spl-token transfer --expected-fee <EXPECTED_FEE> <TOKEN_MINT_ADDRESS> <TOKEN_AMOUNT> <RECIPIENT_WALLET_ADDRESS>
```

It fails with error: `Program TokenzQdBNbLqP5VEhdkAS6EPFLC1PHnBqCXEpPxuEb failed: custom program error: 0x7dc8348c` My suspicion is that TransferCheckedWithFee is not correctly resolving the extra account metas. 

I have already confirmed that this same transfer-hook program works and successfully transfers a token with only the transfer-hook extension.  

Solution: Fix TransferCheckedWithFee to correctly resolve the required extra account metas.

**Comment by joncinque** [comment]
Thanks for your report! This is indeed an issue.

It looks like we'll need to do a few things to fix this properly.

* in token-2022, add an `offchain` helper (and probably an `onchain` helper while we're at it) which also takes in the expected fee, something like `create_transfer_checked_with_fee_instruction_with_extra_metas`, which could resolve to some common helper function with https://github.com/solana-labs/solana-program-library/blob/7d35ad2ed2bc1e6c7f2a841526711c47f5093438/token/program-2022/src/offchain.rs#L37
* in token-client, use that offchain helper in `transfer_with_fee` https://github.com/solana-labs/solana-program-library/blob/7d35ad2ed2bc1e6c7f2a841526711c47f5093438/token/client/src/token.rs#L1149, very similar to how the `transfer` function works https://github.com/solana-labs/solana-program-library/blob/7d35ad2ed2bc1e6c7f2a841526711c47f5093438/token/client/src/token.rs#L992
* token-cli already uses `transfer_with_fee`, so that should Just Work afterwards

**Comment by tonton-sol** [comment]
I will start on this!

**Comment by joncinque** [comment]
Resolved by #7171, thanks!
